crates/dslab-dag/src/tests.rs

```diff
@@ -220,30 +222,49 @@ fn test_4() {
     add_edge(7, 9, 11, "n");
     add_edge(8, 9, 13, "o");
 
-    let mut sim = DagSimulation::new(
-        123,
-        Rc::new(RefCell::new(ConstantBandwidthNetwork::new(1.0, 0.0))),
-        Rc::new(RefCell::new(HeftScheduler::new())),
-        Config {
-            data_transfer_mode: DataTransferMode::Direct,
-        },
-    );
-    sim.add_resource("0", 1, 1, 0);
-    sim.add_resource("1", 2, 1, 0);
-    sim.add_resource("2", 4, 1, 0);
-    sim.add_resource("3", 4, 1, 0);
+    fn run_sim(scheduler: impl crate::scheduler::Scheduler + 'static, dag: DAG) -> f64 {
```

Comment:
Импортируйте Scheduler и переименуйте в `run_scheduler`:

```suggestion
    fn run_scheduler(scheduler: impl Scheduler + 'static, dag: DAG) -> f64 {
```